File: app/src/main/java/xyz/michaelobi/paperplayer/presentation/musiclibrary/MusicLibraryPresenter.java
package xyz.michaelobi.paperplayer.presentation.musiclibrary;

import xyz.michaelobi.paperplayer.event.EventBus;
import xyz.michaelobi.paperplayer.injection.Injector;
import xyz.michaelobi.paperplayer.mvp.BasePresenter;
import xyz.michaelobi.paperplayer.playback.events.action.TogglePlayback;

/**
 * PaperPlayer
 * Michael Obi
 * 15 10 2016 3:45 PM
 */

class MusicLibraryPresenter extends BasePresenter<MusicLibraryContract.View> implements
        MusicLibraryContract.Presenter {
    private EventBus bus;

    public MusicLibraryPresenter() {
        bus = Injector.provideEventBus();
    }

    @Override
    public void attachView(MusicLibraryContract.View view) {
        super.attachView(view);
        initialize();
    }

    public void initialize() {
        checkViewAttached();
        getViewOrThrow().initializeViewPager();
    }

    @Override
    public void onFabClick() {
        bus.post(new TogglePlayback());
    }
}

File: app/src/main/java/xyz/michaelobi/paperplayer/presentation/musiclibrary/MusicLibraryPresenter.kt
/*
 * MIT License
 *
 * Copyright (c) 2017 MIchael Obi
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package xyz.michaelobi.paperplayer.presentation.musiclibrary

import xyz.michaelobi.paperplayer.event.EventBus
import xyz.michaelobi.paperplayer.injection.Injector
import xyz.michaelobi.paperplayer.mvp.BasePresenter
import xyz.michaelobi.paperplayer.playback.events.action.TogglePlayback

/**
 * PaperPlayer
 * Michael Obi
 * 15 10 2016 3:45 PM
 */

internal class MusicLibraryPresenter : BasePresenter<MusicLibraryContract.View>(), MusicLibraryContract.Presenter {
    private val bus: EventBus = Injector.provideEventBus()

    override fun attachView(view: MusicLibraryContract.View) {
        super.attachView(view)
        initialize()
    }

    override fun initialize() {
        checkViewAttached()
        view.initializeViewPager()
    }

    override fun onFabClick() {
        bus.post(TogglePlayback())
    }
}
